portfolio/src/main/java/com/google/sps/servlets/BlobstoreUploadServlet.java

```diff
@@ -32,10 +32,19 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String page = request.getParameter("page");
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
-    String uploadUrl = blobstoreService.createUploadUrl("data");
 
-    response.setContentType("text/html");
-    response.getWriter().println(uploadUrl);
+    // determine which url to generate based on the page passed in
+    if (page.equals("comments")) {
+        String uploadUrl = blobstoreService.createUploadUrl("/data");
```

Comment:
This block of 3 lines is duplicated in the `else`. You can avoid this by doing something like
```
 if (page.equals("comments"))
  endpoint = "/data"
else
  endpoint = "/blog-comment"

String uploadUrl = blobstoreService.createUploadUrl(endpoint);
...
```